src/py/flwr/server/superlink/linkstate/in_memory_linkstate.py

```diff
@@ -215,6 +288,87 @@ def store_task_res(self, task_res: TaskRes) -> Optional[UUID]:
         # Return the new task_id
         return task_id
 
+    # pylint: disable=R0911
+    def store_message_res(self, message: Message) -> Optional[UUID]:
+        """Store one Message."""
+        # Validate task
+        errors = validate_message(message, is_reply_message=True)
+        if any(errors):
+            log(ERROR, errors)
+            return None
+
+        res_metadata = message.metadata
+        with self.lock:
+            # Check if the Message it is replying to exists and is valid
+            msg_ins_id = res_metadata.reply_to_message
+            msg_ins = self.message_ins_store.get(UUID(msg_ins_id))
+
+            # Ensure that dst_node_id of original Message matches the src_node_id of
+            # reply Message.
+            if (
+                msg_ins
+                and message
+                and msg_ins.metadata.dst_node_id != res_metadata.src_node_id
+            ):
+                return None
+
+            if msg_ins is None:
+                log(
+                    ERROR,
+                    "Message with `metadata.message_id` %s does not exist.",
+                    msg_ins_id,
+                )
+                return None
+
+            ins_metadata = msg_ins.metadata
+            if ins_metadata.created_at + ins_metadata.ttl <= time.time():
+                log(
+                    ERROR,
+                    "Failed to store Message: the message it is replying to "
+                    "(with id %s) has expired",
```

Comment:
Should we use uppercase letters here?
```suggestion
                    "(with ID %s) has expired",
```